CMakeLists.txt

```diff
@@ -68,8 +68,13 @@ if(NOT CMAKE_BUILD_TYPE)
     message(STATUS "No CMAKE_BUILD_TYPE specified, defaulting to Debug")
     set(CMAKE_BUILD_TYPE Debug)
 endif(NOT CMAKE_BUILD_TYPE)
+
 set(CMAKE_CXX_FLAGS_DEBUG "-g")
 
+SET(CMAKE_BUILD_TYPE Valgrind)
```

Comment:
I'm not quite sure what this does? It doesn't appear to actually set CMAKE_BUILD_TYPE, but the documentation says that CMAKE_CONFIGURATION_TYPES should be used to add a new configuration type: https://cmake.org/cmake/help/v3.0/variable/CMAKE_CONFIGURATION_TYPES.html

Reply:
I created this configuration type just because on valgrind manual page they claims that if sources has builded with -o1 it improves performance significantly.

http://valgrind.org/docs/manual/quick-start.html